common/modules/farm/contexts/src/generic.rs

```diff
@@ -0,0 +1,432 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use common_structs::FarmTokenAttributes;
+use farm_token::FarmToken;
+
+use config::State;
+
+pub struct StorageCache<M: ManagedTypeApi> {
+    pub contract_state: State,
+    pub farm_token_id: TokenIdentifier<M>,
+    pub farming_token_id: TokenIdentifier<M>,
+    pub reward_token_id: TokenIdentifier<M>,
+    pub reward_reserve: BigUint<M>,
+    pub reward_per_share: BigUint<M>,
+    pub farm_token_supply: BigUint<M>,
+    pub division_safety_constant: BigUint<M>,
+}
```

Comment:
I think it would make more sense to have all fields as `Option<T>` instead. Since, as a cache, it would only actually load what's needed. Then you'd wrap those options in methods, which would check for None and load from storage, or simply return the already stored value.

Reply:
done